src/lib/comms/include/sol-oic-client.h

```diff
@@ -286,13 +287,14 @@ bool sol_oic_client_resource_request(struct sol_oic_client *client, struct sol_o
  *        @ref SOL_OIC_MAP_LOOP() macro.
  * @param data A pointer to user's data.
  * @param observe If server will be obeserved or unobserved.
+ * @param non_confirmable Use non-confirmable CoAP packets to make this request.
  *
  * @return True if packet was successfully sent. False otherwise.
  */
 bool sol_oic_client_resource_set_observable(struct sol_oic_client *client, struct sol_oic_resource *res,
     void (*callback)(struct sol_oic_client *cli, const struct sol_network_link_addr *addr,
     const struct sol_str_slice *href, const struct sol_oic_map_reader *repr_map, void *data),
-    void *data, bool observe);
+    void *data, bool observe, bool non_confirmable);
 
```

Comment:
If at all possible, I'd like to avoid having more than one boolean parameter. From the call site, it's hard to know what `true, false` or `false, true` actually are. For this specific case, I'd create an enum with values `SOL_OIC_OBSERVE = 1<<0` and `SOL_OIC_NONCON = 1<<1` or something along these lines, so that they can be OR-ed in the call site, and thus obvious to the reader